### PR TITLE
HCLOUD-2963_Add-semantic-relese-to-the-wave-nodes-catalog-blueprint_Darius-Weiberg

### PR DESCRIPTION
**Pull request by DariusWe**
Add everything necessary to create changelogs for releases

### CODE REVIEW

File: changelog-howTo.txt
Comment (JorgePBrown) [inline changelog-howTo.txt:1]: Can you change this to a Markdown file and put the JSON inside a code block?

**Review by JorgePBrown — CHANGES_REQUESTED** [review]
Additionally to my other comments, could you one more script which would append to the pre-push hook a check if there are changelog changes when there are node changes?

File: scripts/check-changelog.mjs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    process.exit(1);
+}
+
+const firstEntry = changelog[0];
+
+const semverRegex = /^(\d+)\.(\d+)\.(\d+)$/;
+if (!semverRegex.test(firstEntry.version)) {
+    console.error(`Version ${firstEntry.version} in changelog.json is not a valid version number. Must be 'X.X.X'`);
+    process.exit(1);
+}
```
[the remark this answers is not in the document]
Comment (DariusWe) [reply scripts/check-changelog.mjs:27]: There is an extra ticket for that:
https://app.clickup.com/t/2570196/HCLOUD-3069

I could also do it here though.

**Comment by DariusWe** [comment]
@JorgePBrown Regarding the pre-push hook: These will have no effect in templates. We would have to implement them as custom hooks in the repository and then change the hooksPath. This means that users of the template could not use their own hooks anymore (without changing the hooksPath again), and all contributors would be forced to use the same. Probably not that problematic, but seems not worth if for me for now. What do you think?

File: scripts/check-changelog.mjs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+if (firstEntry.version !== tag) {
+    console.error("Changelog version and pushed tag don't match. You probably forgot to update the changelog");
+    process.exit(1);
+}
+
+const dateRegex = /^\d{4}-\d{2}-\d{2}$/;
+if (!dateRegex.test(firstEntry.date)) {
+    console.error(`Date ${firstEntry.date} is not in YYYY-MM-DD format.`);
+    process.exit(1);
+}
```
[the remark this answers is not in the document]
Comment (DariusWe) [reply scripts/check-changelog.mjs:38]: I will do it, however personally i think my solution is sufficient. Why would we want to allow more formats? The users will just use the format that is defined in the example anyways, and i think from a users perspective a restricted format removes uncertainty about what is allowed and what not.

[the remark this answers is not in the document]
Comment (DariusWe) [reply scripts/check-changelog.mjs]: Done!

File: changelog-howTo.txt
Comment (DariusWe) [reply changelog-howTo.txt:1]: Done!

**Review by JorgePBrown — APPROVED** [review]

**Comment by JorgePBrown** [comment]
> @JorgePBrown Regarding the pre-push hook: These will have no effect in templates. We would have to implement them as custom hooks in the repository and then change the hooksPath. This means that users of the template could not use their own hooks anymore (without changing the hooksPath again), and all contributors would be forced to use the same. Probably not that problematic, but seems not worth if for me for now. What do you think?

Makes sense. Leave it out then

**Pull request merged**